engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/NetworkOrchestrator.java

```diff
@@ -2764,6 +2770,21 @@ public Network doInTransaction(final TransactionStatus status) {
         return network;
     }
 
+    @Override
+    public boolean isSharedNetworkWithoutSpecifyVlan(NetworkOffering offering) {
+        if (offering == null || offering.getTrafficType() != TrafficType.Guest || offering.getGuestType() != GuestType.Shared) {
+            return false;
+        }
+        if (! offering.isSpecifyVlan()) {
```

Comment:
simplify this if checkeck to `return offering.isSpecifyVlan()` ?

Reply:
@ravening done. thanks.

Reply:
@ravening done. thanks.